File: lecture_004(Array)/l_002_maxminvalueinarray.java
import java.util.Scanner;
public class l_002_maxminvalueinarray {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
    	int n = scn.nextInt();
    	int[] arr = new int[n];
    	input(arr);
    	maximum(arr);
    	minimum(arr);
    }
    public static void display(int[] arr) {
    	for(int r: arr) {
    		System.out.print(r+" ");
    	}
    	System.out.println();
    }
    public static void input(int[] arr) {
    	for(int i=0; i<arr.length; i++) {
    		arr[i]=scn.nextInt();
    	}
        display(arr);
    }
    public static void maximum(int[] arr) {
    	int max = arr[0];
    	for(int i=1; i<arr.length; i++) {
    		if(arr[i]>max) {
    			max = arr[i];
    		}
    	}
    	System.out.println("Maximum: "+ max);
    }
    public static void minimum(int[] arr) {
    	int min = arr[0];
    	for(int i=1; i<arr.length; i++) {
    		if(arr[i]<min) {
    			min = arr[i];
    		}
    	}
    	System.out.println("Minimum: "+ min);
    }
}

File: lecture_004(Array)/l_002_maxminvalueinarray.cpp
#include<iostream>
#include<vector>
using namespace std;
void display(vector<int>& arr) {
    for (int i:arr) {
        cout<<i<<" ";
    }
    cout<<endl;
}
void input(vector<int>& arr) {
    for(int i=0; i<arr.size(); i++) {
        cin>>arr[i];
    }
}
void maximum(vector<int>& arr) {
     int max=arr[0];
     for(int i=0;i<arr.size(); i++) {
         if (arr[i]>max) {
             max=arr[i];
         }
     }
     cout<<"Maximum value:"<<max<<endl;
}
void minimum(vector<int>& arr) {
    int min=arr[0];
    for(int i=0; i<arr.size(); i++) {
        if (arr[i]<min) {
            min=arr[i];
        }  
    }
    cout<<"Minimum value:"<<min<<endl; 
}
int main(int args, char**argv) {
    int n;
    cin>>n;
    vector<int>arr(n,0);
    input(arr);
    display(arr);
    maximum(arr);
    minimum(arr);

    return 0;
}
